chainer/functions/math/sign.py

```diff
@@ -0,0 +1,54 @@
+import numpy
+
+from chainer import cuda
+from chainer import function_node
+from chainer import utils
+from chainer.utils import type_check
+
+
+class Sign(function_node.FunctionNode):
+
+    """Element-wise sign function."""
+
+    @property
+    def label(self):
+        return 'sign'
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+
+    def forward_cpu(self, x):
+        return utils.force_array(numpy.sign(x[0])),
+
+    def forward_gpu(self, x):
+        return cuda.cupy.sign(x[0]),
+
+    def backward(self, indexes, grad_outputs):
+        return grad_outputs[0] * 0,
```

Comment:
Why don't you use `zeros_like`?

Reply:
Thank you for the review. Don't we need to return a Variable object here?

Reply:
Sorry, please return `None` in this case.

Reply:
By doing so, double backprop won't work when this function is involved in the computational graph. Please correct me if I am wrong.

Reply:
I came across the same situation (just return zero) in #3539.  I'd like to know if there is a better way. (maybe ask @beam2d ?)

Reply:
Both PyTorch and TensorFlow seem to return 0. This is the code for PyTorch. https://github.com/pytorch/pytorch/blob/3853d5da97ead031d7d77f39fe7ce2de243ce369/tools/autograd/derivatives.yaml#L436-L437